### PR TITLE
import/index: adjust concurrency using cpu-count of managed node depends on use case

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #49008

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/index.go [hunk 1]
```diff
@@ -2118,9 +2117,11 @@ func (w *worker) executeDistTask(reorgInfo *reorgInfo) error {
 
 		job := reorgInfo.Job
 		workerCntLimit := int(variable.GetDDLReorgWorkerCounter())
-		// we're using cpu count of current node, not of framework managed nodes,
-		// but it seems more intuitive.
-		concurrency := min(workerCntLimit, cpu.GetCPUCount())
+		cpuCount, err := handle.GetCPUCountOfManagedNode(ctx)
+		if err != nil {
+			return err
+		}
```
Comment (ywqzzy) [inline pkg/ddl/index.go:2123]: retryable？

File: pkg/executor/importer/import.go [hunk 2]
```diff
@@ -1335,5 +1340,27 @@ func GetMsgFromBRError(err error) string {
...
+		return 0, exeerrors.ErrLoadDataInvalidURI.GenWithStackByArgs(plannercore.ImportIntoDataSource,
+			err2.Error())
+	}
+
+	serverDiskImport := storage.IsLocal(u)
+	if serverDiskImport || !variable.EnableDistTask.Load() {
+		return cpu.GetCPUCount(), nil
+	}
+	// the call path of initialization of threadCnt don't have context, so we use a timeout here.
+	ctx, cancel := context.WithTimeout(context.Background(), getCPUCountTimeout)
```
Comment (ywqzzy) [inline pkg/executor/importer/import.go:1359]: What if have a heavy workload in CI？

Comment (D3Hunter) [reply pkg/executor/importer/import.go:1359]: we can refactor the call path to pass a context, too lazy to change them, will think about it tomorrow

Comment (D3Hunter) [reply pkg/executor/importer/import.go]: https://github.com/pingcap/tidb/pull/50091/commits/058bb613a828b1ae87847fe412e3010f51f5af9f

**Comment by D3Hunter** [comment]
cc @ywqzzy @tangenta

File: pkg/disttask/framework/storage/task_table.go [hunk 3]
```diff
@@ -1257,5 +1269,8 @@ func (stm *TaskManager) getCPUCountOfManagedNodes(ctx context.Context, se sessio
 			break
 		}
 	}
+	if cpuCount == 0 {
+		return 0, errors.New("no managed node have enough resource")
```
Comment (ywqzzy) [inline pkg/disttask/framework/storage/task_table.go:1273]:
```suggestion
		return 0, errors.New("no managed node have enough resource for dist task")
```

File: pkg/executor/importer/import.go [hunk 4]
```diff
@@ -141,6 +143,8 @@ var (
 		".zstd", ".zst",
 		".snappy",
 	}
+
+	getCPUCountTimeout = time.Minute
```
Comment (ywqzzy) [inline pkg/executor/importer/import.go:147]:
```suggestion
```

File: pkg/executor/importer/importer_testkit_test.go [hunk 5]
```diff
@@ -65,3 +68,34 @@ func TestChecksumTable(t *testing.T) {
...
+func TestGetTargetNodeCpuCnt(t *testing.T) {
+	_, tm, ctx := testutil.InitTableTest(t)
+	require.False(t, variable.EnableDistTask.Load())
+
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/util/cpu/mockNumCpu", "return(16)"))
+	t.Cleanup(func() {
+		require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/pkg/util/cpu/mockNumCpu"))
+		variable.EnableDistTask.Store(false)
+	})
+	require.NoError(t, tm.StartManager(ctx, "tidb1", ""))
```
Comment (ywqzzy) [inline pkg/executor/importer/importer_testkit_test.go:81]: fix confilt

**Review by ywqzzy — APPROVED** [review]
Rest LGTM

**Review by tangenta — APPROVED** [review]

**Pull request merged**